### PR TITLE
ROP pipeline prototype

### PR DESCRIPTION
**Pull request by arunsurya77**
ROP pipeline prototype implementation.  Implements sampling and non-linearity correction. Currently, non-linearity correction coefficients need to be supplied.  Sampling code requires drsrop_clib, currently a separate repo. Documentation in drsrop_pipeline.rst (not linked)

### CODE REVIEW

**Comment by zonca** [comment]
thanks @arunsurya77 you can use git submodule https://git-scm.com/book/en/v2/Git-Tools-Submodules and add the Python and Cython code to `iris_pipeline` and import the C library from `iris_readout`

**Comment by zonca** [comment]
Please let me know when this is ready for review

**Comment by arunsurya77** [comment]
I have the C library inside the iris_pipeline in my fork now.  I am
currently working to get the c lib as a separate repo as a submodule. Doing
some tweaking for this. i will anyway give a review request.

**Comment by zonca** [comment]
@arunsurya77 I made some updates to `iris_pipeline` so there are some conflicts now, also make sure there are some unit tests that automatically run. I currently have some tests which are Jupyter Notebooks, I like them because they are easy to run interactively and to debug anything gone wrong, it also includes some plotting, see for example: https://github.com/oirlab/iris_pipeline/blob/master/iris_pipeline/tests/run_image2_subarray.ipynb

**Comment by arunsurya77** [comment]
Hi Andrea, I will check the merge conflicts and try to resolve them and add
the tests.

File: docs/drsrop_pipeline.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+       mode='mcds'
+        
+
+The sampling mode is set by the ``mode`` keyword which can be ``mcds`` or ``utr``. MCDS algorithm also requires the group number, the number of reads to be co-added. This is currently hardcoded in this version.
+
+
+Execute the pipeline from the command line
+------------------------------------------
+
+We can use ``strun`` from a terminal to execute the pipeline:
```
Comment (zonca) [inline docs/drsrop_pipeline.rst]: we have `tmtrun`

**Comment by zonca** [comment]
@arunsurya77 what is the stage of this?
I think this should be high priority, it would help a lot to have a clearer picture of all features of `iris_pipeline` and improve the documentation.

**Comment by arunsurya77** [comment]
Yeah I agree, sorry about the delay. I hadn't had time to work on this. I will try to fix some of the outstanding issues this week and update the rst file of this. There are some problems I am facing with the monkey patching. But I will atleast try to do push all the updates so that you can have a look.

File: docs/drsrop_pipeline.rst (hunk 1 above)
Comment (zonca) [reply docs/drsrop_pipeline.rst:51]: @arunsurya77 you should use `tmtrun` which automatically monkeypatches `strun`

**Comment by zonca** [comment]
@arunsurya77 do you confirm this code is currently working on your machine?

**Comment by arunsurya77** [comment]
High I'm just editing the last bits for the rst documentation. I'll make a
pull request tonight. It's just a few more changes but everything is
working, I managed to fix the erros.

**Comment by arunsurya77** [comment]
I have added most of the ROP Pipeline Skelton in this pull request. I have verified it working in another ubuntu system also. Just to give you the installation process. I have documented it in the rop_pipeline.rst file also.

requirements --
libcfitsio  [ libcfitsio-bin and libcfitsio-dev (sudo apt-get install libcfitsio*)]
cython

download 'sample_ramp_new.fits<' https://figshare.com/articles/sample_ramp_new_fits/12462491>  to iris_pipeline/tests/data/ and iris_pipeline/readout/tests/ . 

The pytest is currently minimal which just runs the ROP pipleine on the data file. I am adding more tests and some frames from the simulator.

**Comment by zonca** [comment]
let's switch to https://github.com/oirlab/iris_pipeline/pull/27 so I can help edit this.

**Pull request closed without merging**